framework/src/org/checkerframework/common/value/util/Range.java

```diff
@@ -767,6 +767,38 @@ public Range refineEqualTo(Range right) {
     }
 
     /**
+     * Refines this range to reflect that no value in it can be equal to a value in the given range.
+     * This is used for calculating the control-flow-refined result of the != operator. For example:
+     *
+     * <pre>
+     * <code>
+     *     {@literal @}IntRange(from = 0, to = 10) int a;
+     *     {@literal @}IntRange(from = 3, to = 15) int b;
+     *     ...
+     *     if (a != b) {
+     *         // range of <i>a</i> is now refined to [0, 2] because a value in range [3, 10]
```

Comment:
Haha no worries! I was just working on the other refinement stuff for the Value Checker, and so this seemed like a natural extension. Plus it'll be really helpful for some of the false positives we're seeing commonly in the Index Checker.

I also didn't actually do much besides call out to your code and write tests. Thanks for writing such a good Range library.